addons/stock/models/stock_quant.py

```diff
@@ -249,7 +249,7 @@ def check_product_id(self):
     def check_quantity(self):
         for quant in self:
             if float_compare(quant.quantity, 1, precision_rounding=quant.product_uom_id.rounding) > 0 and quant.lot_id and quant.product_id.tracking == 'serial':
-                raise ValidationError(_('A serial number should only be linked to a single product.'))
+                raise ValidationError(_('The serial number "%s" for product "%s" has already been assigned.' % (quant.lot_id.name, quant.product_id.display_name)))
```

Comment:
don't need to translate lot name/product name

Reply:
Then it should be like this?
`_('The serial number ') + quant.lot_id.name + _(' for product ') + quant.product_id.display_name + _(' has already been assigned. ')`
It looks wired and I don't know if it can be well translated.

Reply:
Maybe I should display the product and lot in another line?